### PR TITLE
refactor: [M3-7006] - Make `UserMenu` use MUI instead of Reach UI

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝

- The codebase has many different packages that all do the same things, we are trying to consolidate. In this case, we want to use MUI's Menu/Popper/Popover components instead of Reach UI's Menu
- This is similar to https://github.com/linode/manager/pull/9224

## Major Changes 🔄
- Makes `UserMenu` use a MUI `Popover` instead of a Reach UI `Menu`
- Uses **a ton** less one-off styles

## Preview 📷

| Before  | After   |
| ------- | ------- |
| Screenshot 2023-08-10 at 6 51 46 PM | Screenshot 2023-08-10 at 6 51 52 PM |

- Verify the `UserMenu` still works as expected and looks similar to the old one

### CODE REVIEW

File: packages/manager/src/features/NotificationCenter/NotificationSection.tsx [hunk 1]
```diff
@@ -91,11 +87,7 @@ export const NotificationSection = (props: NotificationSectionProps) => {
                   <Typography variant="h3">{header}</Typography>
                   {showMoreTarget && (
                     <strong>
-                      <Link
-                        className={classes.menuItemLink}
-                        style={{ padding: 0 }}
-                        to={showMoreTarget}
-                      >
+                      <Link style={{ padding: 0 }} to={showMoreTarget}>
```
Comment (abailly-akamai) [inline packages/manager/src/features/NotificationCenter/NotificationSection.tsx:90]: nitpick: use `sx`

**Review by abailly-akamai — APPROVED** [review]
Nice! I think accessibility is actually improved as well - may have had to do with previous implementation but tabbing on open seems to work better now 👍 

Approving pending a couple defects that should be addressed

Screenshot 2023-08-11 at 14 38 18

Screenshot 2023-08-11 at 14 38 09

**Review by jaalah-akamai — APPROVED** [review]

File: packages/manager/src/features/NotificationCenter/NotificationSection.tsx (hunk 1 above)
Comment (bnussman-akamai) [reply packages/manager/src/features/NotificationCenter/NotificationSection.tsx:90]: no `sx` on the `Link` component 😖

Comment (abailly-akamai) [reply packages/manager/src/features/NotificationCenter/NotificationSection.tsx:90]: ah.. could be added but no big deal, can be a follow up

**Pull request merged**